modules/siddhi-extensions/math/src/main/java/org/wso2/siddhi/extension/math/PercentileFunctionExtension.java

```diff
@@ -0,0 +1,394 @@
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.siddhi.extension.math;
+
+import org.wso2.siddhi.core.config.ExecutionPlanContext;
+import org.wso2.siddhi.core.exception.ExecutionPlanCreationException;
+import org.wso2.siddhi.core.exception.OperationNotSupportedException;
+import org.wso2.siddhi.core.executor.ConstantExpressionExecutor;
+import org.wso2.siddhi.core.executor.ExpressionExecutor;
+import org.wso2.siddhi.query.api.definition.Attribute;
+import org.wso2.siddhi.core.query.selector.attribute.aggregator.AttributeAggregator;
+
+import org.apache.commons.math3.stat.descriptive.DescriptiveStatistics;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * percentile(values, p)
+ * Returns an estimate for the pth percentile of the stored values
+ * Accept Type(s): FLOAT,INT,LONG,DOUBLE
+ * Return Type: DOUBLE
+ */
+public class PercentileFunctionExtension extends AttributeAggregator {
+
+    private PercentileFunctionExtension percentileOutputFunctionExtension;
+    private double percentileValue;
+
+    /**
+     * The initialization method for FunctionExecutor
+     *
+     * @param attributeExpressionExecutors are the executors of each attributes in the function
+     * @param executionPlanContext Execution plan runtime context
+     */
+    @Override
+    protected void init(ExpressionExecutor[] attributeExpressionExecutors, ExecutionPlanContext executionPlanContext) {
+
+        if (attributeExpressionExecutors.length != 2) {
+            throw new OperationNotSupportedException("Percentile function has to have exactly 2 parameter, currently "
+                    + attributeExpressionExecutors.length + " parameters provided");
+        }
+
+        try {
+
+            if (attributeExpressionExecutors[1] instanceof ConstantExpressionExecutor) {
+                percentileValue = ((Double) attributeExpressionExecutors[1].execute(null));
+            } else {
+                throw new OperationNotSupportedException("Percentile value has to be a constant");
+            }
+
+        } catch (ClassCastException c) {
+            throw new ExecutionPlanCreationException("Percentile value should be of type double. But found "
+                    + attributeExpressionExecutors[1].getReturnType());
+        }
+
+        if (percentileValue <= 0 || percentileValue > 100) {
+            throw new OperationNotSupportedException(
+                    "Percentile value should be in 0 < p ≤ 100 range. But found " + percentileValue);
+        }
+
+        Attribute.Type type = attributeExpressionExecutors[0].getReturnType();
+
+        switch (type) {
+        case FLOAT:
+            percentileOutputFunctionExtension = new PercentileFunctionExtensionFloat();
```

Comment:
Well, we need to know the type at some point since we are using addValue method[1] in Apache commons DescriptiveStatistics library.
[1][http://commons.apache.org/proper/commons-math/apidocs/org/apache/commons/math3/stat/descriptive/DescriptiveStatistics.html#addValue(double)](url)